src/main/java/uk/gov/companieshouse/itemhandler/kafka/OrdersKafkaConsumer.java

```diff
@@ -1,173 +1,248 @@
+            throws InterruptedException, ExecutionException, SerializationException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
         try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY);
-        } catch (Exception ex){
-            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, ex);
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC_RETRY);
+
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC_RETRY);
+        } catch (ServiceException ex){
+            logMessageProcessingFailureRecoverable(message, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, ex);
+            republishMessageToTopic(orderReceivedUri, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR);
+        } catch (Exception x) {
+            logMessageProcessingFailureNonRecoverable(message, ORDER_RECEIVED_TOPIC_RETRY, ORDER_RECEIVED_TOPIC_ERROR, x);
         }
     }
 
-    @KafkaListener(topics = ORDER_RECEIVED_TOPIC_ERROR, groupId = ORDER_RECEIVED_GROUP_ERROR,
-            autoStartup = "${uk.gov.companieshouse.item-handler.error-consumer}")
-    public void processOrderReceivedError(String orderReceivedUri) {
+    /**
+     * `order-received-error` topic listener/consumer is enabled when the application is launched in error
+     * mode (IS_ERROR_QUEUE_CONSUMER=true). Receives messages up to `ERROR_RECOVERY_OFFSET` offset. Calls a `retryable`
+     * method to process received message. If the `retryable` processor is unsuccessful with a `retryable` error, after
+     * maximum numbers of attempts allowed, the message is republished to `-retry` topic for failover processing.
+     * This listener stops accepting messages when the topic's offset reaches `ERROR_RECOVERY_OFFSET`.
+     * @param message
+     * @throws SerializationException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @KafkaListener(id = ORDER_RECEIVED_GROUP_ERROR, groupId = ORDER_RECEIVED_GROUP_ERROR,
+                    topics = ORDER_RECEIVED_TOPIC_ERROR,
+                    autoStartup = "${uk.gov.companieshouse.item-handler.error-consumer}")
+    public void processOrderReceivedError(org.springframework.messaging.Message<OrderReceived> message)
+            throws SerializationException, ExecutionException, InterruptedException {
+        OrderReceived orderReceived = message.getPayload();
+        String orderReceivedUri = orderReceived.getOrderUri();
         try {
-            logMessageReceived(orderReceivedUri, ORDER_RECEIVED_TOPIC_ERROR);
-        } catch (Exception x){
-            LOGGER.error("Message: " + orderReceivedUri + " received on topic: " + ORDER_RECEIVED_TOPIC_ERROR + " could not be processed.");
+            logMessageReceived(message, ORDER_RECEIVED_TOPIC_ERROR);
+
+            logMessageProcessed(message, ORDER_RECEIVED_TOPIC_ERROR);
```

Comment:
Message appears to be logged twice.

Reply:
They are two separate messages. The processing of received message (call to retryable method that sends email) happens between these two messages.